cloud/google/machineactuator_test.go

```diff
@@ -0,0 +1,119 @@
+package google_test
+
+import (
+	"github.com/stretchr/testify/assert"
+	compute "google.golang.org/api/compute/v1"
+	"sigs.k8s.io/cluster-api/cloud/google"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"strings"
+	"testing"
+)
+
+type GCEClientComputeServiceMock struct {
+	mockImagesGet           func(project string, image string) (*compute.Image, error)
+	mockImagesGetFromFamily func(project string, family string) (*compute.Image, error)
+	mockInstancesDelete     func(project string, zone string, targetInstance string) (*compute.Operation, error)
+	mockInstancesGet        func(project string, zone string, instance string) (*compute.Instance, error)
+	mockInstancesInsert     func(project string, zone string, instance *compute.Instance) (*compute.Operation, error)
+	mockZoneOperationsGet   func(project string, zone string, operation string) (*compute.Operation, error)
+}
+
+func (c *GCEClientComputeServiceMock) ImagesGet(project string, image string) (*compute.Image, error) {
+	if c.mockImagesGet == nil {
+		return nil, nil
+	}
+	return c.mockImagesGet(project, image)
+}
+
+func (c *GCEClientComputeServiceMock) ImagesGetFromFamily(project string, family string) (*compute.Image, error) {
+	if c.mockImagesGetFromFamily == nil {
+		return nil, nil
+	}
+	return c.mockImagesGetFromFamily(project, family)
+}
+
+func (c *GCEClientComputeServiceMock) InstancesDelete(project string, zone string, targetInstance string) (*compute.Operation, error) {
+	if c.mockInstancesDelete == nil {
+		return nil, nil
+	}
+	return c.mockInstancesDelete(project, zone, targetInstance)
+}
+
+func (c *GCEClientComputeServiceMock) InstancesGet(project string, zone string, instance string) (*compute.Instance, error) {
+	if c.mockInstancesGet == nil {
+		return nil, nil
+	}
+	return c.mockInstancesGet(project, zone, instance)
+}
+
+func (c *GCEClientComputeServiceMock) InstancesInsert(project string, zone string, instance *compute.Instance) (*compute.Operation, error) {
+	if c.mockInstancesInsert == nil {
+		return nil, nil
+	}
+	return c.mockInstancesInsert(project, zone, instance)
+}
+
+func (c *GCEClientComputeServiceMock) ZoneOperationsGet(project string, zone string, operation string) (*compute.Operation, error) {
+	if c.mockZoneOperationsGet == nil {
+		return nil, nil
+	}
+	return c.mockZoneOperationsGet(project, zone, operation)
+}
+
```

Comment:
I would suggest creating the cluster and machine in the code instead of parsing yaml files. Parsing is the CLI tool's job and not the machine actuator's job. Plus, it is easier to test more dynamic scenarios when the objects are not hardcoded in a file somewhere.

Reply:
There's some complexity here in that "machineactuator.go" expects to decode() the raw bytes of machine.Spec.ProviderConfig. For that reason I was trying to use yaml to setup various scenarios as it requires less changes. 

I'm experimenting with some other ideas where the decoding() is decoupled from machineactuator.go. I will likely update the PR with a different approach.